transformations/noun_compound_paraphraser/transformation.py

```diff
@@ -0,0 +1,78 @@
+import random
+
+import os
+import json
+
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+
+
+def noun_compound_paraphraser(text, compounds):
+
+    L = [[text]]
+    for k in compounds.keys():
+        if k in L[0][0]:
+            new = []
+            for l in L:
+                for text in l:
+                    n = [text.replace(k,paraphrase[0]) for paraphrase in compounds[k]]
+                    new.extend(n)
+            L.append(new)
+    # Flatten the list and ignore the first (the original text)
+    paraphrases = [x for u in L[1:] for x in u]
+
+    return paraphrases
+
+
+class NounCompoundParaphraser(SentenceOperation):
+    """Replaces two-word noun compounds with a paraphrase.
+
+    Args:
+        max_paraphrases: Maximimum number of paraphrases per noun compound.
+            All combinations of paraphrases will be returned whenever more
+            than one noun compound from the dictionary is found in the sentence.
+            Default: 1.
+        seed: initial seed. Defaults: 0.
+    """
+
+    tasks = [
+        TaskType.TEXT_CLASSIFICATION,
+        TaskType.TEXT_TO_TEXT_GENERATION,
+        TaskType.TEXT_TAGGING,
```

Comment:
The TaskType.TEXT_TAGGING doesn't seem relevant, since the is no tag tokens in the inputs or outputs.